api/v01/vrp.rb

```diff
@@ -536,33 +536,41 @@ def self.vrp_request_schedule(this)
             begin
               # Api key is not declared as part of the VRP and must be handled carefully and separatly from other parameters
               api_key = params[:api_key]
-              checksum = Digest::MD5.hexdigest Marshal.dump(params)
               d_params = declared(params, include_missing: false)
               vrp_params = d_params[:points] ? d_params : d_params[:vrp]
-              APIBase.dump_vrp_dir.write([api_key, vrp_params[:name], checksum].compact.join('_'), { vrp: vrp_params }.to_json) if OptimizerWrapper.config[:dump][:vrp]
 
-              APIBase.services(api_key)[:params_limit].merge(OptimizerWrapper.access[api_key][:params_limit] || {}).each{ |key, value|
+              APIBase.profile(api_key)[:params_limit].merge(OptimizerWrapper.access[api_key][:params_limit] || {}).each{ |key, value|
                 next if vrp_params[key].nil? || value.nil? || vrp_params[key].size <= value
 
                 error!({
                   status: 'Exceeded params limit',
-                  message: "Exceeded #{key} limit authorized for your account: #{value}. Please contact support or sales to increase limits."
-                }, 400)
+                  message: "Exceeded \"#{key}\" limit authorized for your account: #{value}. Please contact support or sales to increase limits."
+                }, 413)
               }
 
+              count :submit_vrp, true, params[:vrp][:points].count
+
+              checksum = Digest::MD5.hexdigest Marshal.dump(params)
```

Comment:
why do you move these lines?

Reply:
please answer to question instead of just resolving thread and keeping diff in new version of the PR. How TU can be OK since you need `checksum` on line 546/547?